File: project/jimmer-core/src/test/java/org/babyfish/jimmer/RegisterRequestTest.java
package org.babyfish.jimmer;

import org.babyfish.jimmer.model.RegisterRequest;
import org.babyfish.jimmer.model.RegisterRequestDraft;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import javax.validation.ValidationException;

public class RegisterRequestTest {

    @Test
    public void testSetIdCard() {
        RegisterRequest request = RegisterRequestDraft.$.produce(draft -> {
            draft.setIdCard("123456789123456789");
        });
        Assertions.assertEquals(
                "{\"idCard\":\"123456789123456789\"}",
                request.toString()
        );
    }

    @Test
    public void testSetIdCardFailed() {
        ValidationException ex = Assertions.assertThrowsExactly(ValidationException.class, () -> {
            RegisterRequestDraft.$.produce(draft -> {
                draft.setIdCard("1234567891234567");
            });
        });
        Assertions.assertEquals("The id-card number is illegal", ex.getMessage());
    }

    @Test
    public void setSetPassword() {
        RegisterRequest request = RegisterRequestDraft.$.produce(draft -> {
            draft.setPassword("abc");
            draft.setPasswordAgain("abc");
        });
        Assertions.assertEquals(
                "{\"password\":\"abc\",\"passwordAgain\":\"abc\"}",
                request.toString()
        );
    }

    @Test
    public void testSetPasswordFailed() {
        ValidationException ex = Assertions.assertThrowsExactly(ValidationException.class, () -> {
            RegisterRequestDraft.$.produce(draft -> {
                draft.setPassword("abc");
                draft.setPasswordAgain("abC");
            });
        });
        Assertions.assertEquals("Two passwords must be same", ex.getMessage());
    }
}

File: project/jimmer-core-kotlin/src/test/kotlin/org/babyfish/jimmer/kt/RegisterRequestTest.kt
package org.babyfish.jimmer.kt

import org.babyfish.jimmer.kt.model.RegisterRequest
import org.babyfish.jimmer.kt.model.by
import org.junit.Test
import java.lang.IllegalArgumentException
import javax.validation.ValidationException
import kotlin.test.assertFailsWith
import kotlin.test.expect

class RegisterRequestTest {

    @Test
    fun testSetIdCard() {
        val request = new(RegisterRequest::class).by {
            idCard = "123456789123456789"
        }
        expect("""{"idCard":"123456789123456789"}""") {
            request.toString()
        }
    }

    @Test
    fun testSetIdCardFailed() {
        assertFailsWith(
            ValidationException::class
        ) {
            new(RegisterRequest::class).by {
                idCard = "12345678912345678"
            }
        }.let {
            expect("Illegal id-card number") {
                it.message
            }
        }
    }

    @Test
    fun testSetPassword() {
        val request = new(RegisterRequest::class).by {
            password = "abc"
            passwordAgain = "abc"
        }
        expect("""{"password":"abc","passwordAgain":"abc"}""") {
            request.toString()
        }
    }

    @Test
    fun testSetPasswordFailed() {
        assertFailsWith(
            ValidationException::class
        ) {
            new(RegisterRequest::class).by {
                password = "abc"
                passwordAgain = "abC"
            }
        }.let {
            expect("The passwords must be same") {
                it.message
            }
        }
    }
}